it/controllers/benefits/reimbursed/NonTaxableCostsBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,528 @@
+            formPostLinkCheck(formPostLink, formSelector)
+
+            errorAboveElementCheck(user.specificExpectedResults.get.expectedOverMaximumErrorMessage, Some(amountFieldName))
+            errorSummaryCheck(user.specificExpectedResults.get.expectedOverMaximumErrorMessage, amountFieldHref)
+
+            welshToggleCheck(user.isWelsh)
+
+          }
+        }
+      }
+    }
+
+    val newAmount: BigDecimal = 500.55
+
+    "update cya when a user submits a valid form and has prior benefits" which {
+
+      val form: Map[String, String] = Map(AmountForm.amount -> newAmount.toString())
+
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
+        insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel("name", hmrc = true,
+          Some(benefits(Some(fullReimbursedCostsVouchersAndNonCashModel))))), userRequest)
+        urlPost(nonTaxableCostsBenefitsAmountPageUrl(taxYearEOY), follow = false, body = form,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      s"redirect to check your benefits page" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+
+      s"update expenses in reimbursedCostsVouchersAndNonCash model" in {
+        lazy val cyaModel = findCyaData(taxYearEOY, employmentId, userRequest).get
+        cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.reimbursedCostsVouchersAndNonCashQuestion)) shouldBe Some(true)
+        cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expensesQuestion)) shouldBe Some(true)
+        cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expenses)) shouldBe Some(newAmount)
+      }
+
+    }
+
+    "update cya when a user submits a valid form and doesn't have prior benefits" which {
+
+      val form: Map[String, String] = Map(AmountForm.amount -> newAmount.toString())
+
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        insertCyaData(employmentUserData(hasPriorBenefits = false, cyaModel("name", hmrc = true,
+          Some(benefits(Some(fullReimbursedCostsVouchersAndNonCashModel))))), userRequest)
+        urlPost(nonTaxableCostsBenefitsAmountPageUrl(taxYearEOY), follow = false, body = form,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      s"redirect to taxable expenses yes/no question page" in {
+        result.status shouldBe SEE_OTHER
+        //TODO - redirect to taxable expenses yes no question page when available
```

Comment:
again you can update the controller to hit the real controller, so can update this here too

Reply:
done